packages/teams-js/src/public/meeting.ts

```diff
@@ -334,6 +334,47 @@ export namespace meeting {
      * @returns A promise with the updated microphone state
      */
     micMuteStateChangedCallback: (micState: MicState) => Promise<MicState>;
+    /**
+     * Callback for the host to tell the app to change its speaker selection
+     */
+    audioDeviceSelectionChangedCallback?: (selectedDevices: AudioDeviceSelection) => Promise<void>;
+  }
+
+  /**
+   * Interface for AudioDeviceSelection from Teams selection
+   *
+   * @hidden
+   * Hide from docs.
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   *
+   * @beta
+   */
+  export interface AudioDeviceSelection {
+    speaker?: AudioDeviceInfo;
+    microphone?: AudioDeviceInfo;
+    error?: SdkError;
+  }
```

Comment:
Is it correct that these could all be `undefined`? How should the developer interpret the situation where the `audioDeviceSelectionChangedCallback` is called and the object passed as the `selectedDevices` parameter has no defined properties on it?

Reply:
From API design aspect, `teams-js` can't be sure these will always have values, Teams might have a bug, or the user don't have any audio I/O devices. Developer can use the system's default device if needed like this example:
```ts
const devices = await navigator.mediaDevices.enumerateDevices();
const defaultSpeaker = devices.find((d) => d.deviceId === 'default' && d.kind === 'audiooutput');
const defaultMic = devices.find((d) => d.deviceId === 'default' && d.kind === 'audioinput');
```